### ISSUE TITLE
Cannot find source file (a long path with unicode) when upload file to googledrive

### ISSUE THREAD

**Issue by hagemouse** [rclone/rclone#5144]
When I copy a file from my disk to google drive, error occurred:
(win2008,rclone 1.54.1)

Command:

```
C:\Users\Administrator>C:\rclone\rclone copy "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4" "baiduyun:baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表/" -vv 
```

Output:

```
2021/03/19 11:41:08 DEBUG : rclone: Version "v1.54.1" starting with parameters ["C:\\rclone\\rclone" "copy" "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4" "baiduyun:baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表/" "-vv"]

2021/03/19 11:41:08 DEBUG : Creating backend with remote "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4"
2021/03/19 11:41:08 DEBUG : Using config file from "C:\\Users\\Administrator\\.config\\rclone\\rclone.conf"

2021/03/19 11:41:08 DEBUG : fs cache: renaming cache item "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4" to be canonical "//?/C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4"

2021/03/19 11:41:08 DEBUG : Creating backend with remote "baiduyun:baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表/"

2021/03/19 11:41:13 DEBUG : fs cache: renaming cache item "baiduyun:baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表/" to be canonical "baiduyun:baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表"

2021/03/19 11:41:13 ERROR : : error reading source directory: directory not found

2021/03/19 11:41:13 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for checks to finish
2021/03/19 11:41:13 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for transfers to finish
2021/03/19 11:41:13 ERROR : Attempt 1/3 failed with 1 errors and: directory not found
2021/03/19 11:41:14 ERROR : : error reading source directory: directory not found
2021/03/19 11:41:14 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for checks to finish
2021/03/19 11:41:14 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for transfers to finish
2021/03/19 11:41:14 ERROR : Attempt 2/3 failed with 1 errors and: directory not found
2021/03/19 11:41:14 ERROR : : error reading source directory: directory not found
2021/03/19 11:41:14 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for checks to finish
2021/03/19 11:41:14 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for transfers to finish
2021/03/19 11:41:14 ERROR : Attempt 3/3 failed with 1 errors and: directory not found
2021/03/19 11:41:14 INFO  :
Transferred:             0 / 0 Bytes, -, 0 Bytes/s, ETA -
Errors:                 1 (retrying may help)
Elapsed time:         5.7s

2021/03/19 11:41:14 DEBUG : 5 go routines active
2021/03/19 11:41:14 Failed to copy: directory not found
```

But the file is in the directory.
Is there any rules in the source path, such as unicode or special chars, or the length?
But some files are ok.

Thank you very much.

**Comment by darthShadow** [comment]
Can you try with the latest beta and see if that works any better?

**Comment by hagemouse** [comment]
I tried the beta version, but still didn't work. I changed "/" to "\", and used "dir" to test if it is exist.

Dir result:

C:\Users\Administrator>dir "C:\BaiduNetdiskDownload\Book\160.秋叶Excel数据处理学习班（完结）\第07课：图表篇\【7-2】切片器是动态统计和动态图表\31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4"
 驱动器 C 中的卷没有标签。


 C:\BaiduNetdiskDownload\Book\160.秋叶Excel数据处理学习班（完结）\第07课：图表篇\【7-2】切片器是动态统计和动态图表 的目录

2021/03/19  11:16               333 31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4
               1 个文件            333 字节
               0 个目录 28,771,020,800 可用字节


rclone 1.55:

C:\Users\Administrator>C:\rclone\rclone copy "C:\BaiduNetdiskDownload\Book\160.秋叶Excel数据处理学习班（完结）\第07课：图表篇\【7-2】切片器是动态统计和动态图表\31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4" "baiduyun:baiduyun\Book\160.秋叶Excel数据处理学习班（完结）\第07课：图表篇\【7-2】切片器是动态统计和动态图表/" -vv
2021/03/20 06:22:29 DEBUG : Using config file from "C:\\Users\\Administrator\\.config\\rclone\\rclone.conf"
2021/03/20 06:22:29 DEBUG : rclone: Version "v1.55.0-beta.5318.2bdbf00fa" starting with parameters ["C:\\rclone\\rclone" "copy" "C:\\BaiduNetdiskDownload\\Book\\160.秋叶Excel数据处理学习班（完结）\\第07课：图表篇\\【7-2】切片器是动态统计和动态图表\\31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4" "baiduyun:baiduyun\\Book\\160.秋叶Excel数据处理学习班（完结）\\第07课：图表篇\\【7-2】切片器是动态统计和动态图表/" "-vv"]

2021/03/20 06:22:29 DEBUG : Creating backend with remote "C:\\BaiduNetdiskDownload\\Book\\160.秋叶Excel数据处理学习班（完结）\\第07课：图表篇\\【7-2】切片器是动态统计和动态图表\\31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4
"
2021/03/20 06:22:29 DEBUG : fs cache: renaming cache item "C:\\BaiduNetdiskDownload\\Book\\160.秋叶Excel数据处理学习班（完结）\\第07课：图表篇\\【7-2】切片器是动态统计和动态图表\\31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4" to be canonical "//?/C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4"
2021/03/20 06:22:29 DEBUG : Creating backend with remote "baiduyun:baiduyun\\Book\\160.秋叶Excel数据处理学习班（完结）\\第07课：图表篇\\【7-2】切片器是动态统计和动态图表/"
2021/03/20 06:22:31 DEBUG : fs cache: renaming cache item "baiduyun:baiduyun\\Book\\160.秋叶Excel数据处理学习班（完结）\\第07课：图表篇\\【7-2】切片器是动态统计和动态图表/" to be canonical "baiduyun:baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表"
2021/03/20 06:22:31 ERROR : : error reading source directory: directory not found
2021/03/20 06:22:31 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for checks to finish
2021/03/20 06:22:31 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for transfers to finish
2021/03/20 06:22:31 ERROR : Attempt 1/3 failed with 1 errors and: directory not found
2021/03/20 06:22:31 ERROR : : error reading source directory: directory not found
2021/03/20 06:22:31 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for checks to finish
2021/03/20 06:22:31 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for transfers to finish
2021/03/20 06:22:31 ERROR : Attempt 2/3 failed with 1 errors and: directory notfound
2021/03/20 06:22:32 ERROR : : error reading source directory: directory not found
2021/03/20 06:22:32 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for checks to finish
2021/03/20 06:22:32 DEBUG : Google drive root 'baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表': Waiting for transfers to finish
2021/03/20 06:22:32 ERROR : Attempt 3/3 failed with 1 errors and: directory not found
2021/03/20 06:22:32 INFO  :
Transferred:             0 / 0 Bytes, -, 0 Bytes/s, ETA -
Errors:                 1 (retrying may help)
Elapsed time:         2.8s

2021/03/20 06:22:32 DEBUG : 5 go routines active
2021/03/20 06:22:32 Failed to copy: directory not found

**Comment by ivandeex** [comment]
Hmm. Works for me with rclone 1.54.0, Windows 8.1 Rus and GoogleDrive with file names copy-pasted from this page to the windows console (I use ConEmu with PowerShell but also tested in Vanilla CMD).


rclone copyto -vv ./delluc.mp4 "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4"
2021/03/20 14:43:53 DEBUG : rclone: Version "v1.54-iva01" starting with parameters ["C:\\abyss\\bin\\rclone.exe" "copyto" "-vv" "./delluc.mp4" "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4"]
2021/03/20 14:43:53 DEBUG : Creating backend with remote "./delluc.mp4"
2021/03/20 14:43:53 DEBUG : Using config file from "C:\\Users\\ivandeex\\.config\\rclone\\rclone.conf"
2021/03/20 14:43:53 DEBUG : fs cache: adding new entry for parent of "./delluc.mp4", "//?/C:/abyss/temp"
2021/03/20 14:43:53 DEBUG : Creating backend with remote "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/"
2021/03/20 14:43:53 DEBUG : fs cache: renaming cache item "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/" to be canonical "//?/C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/"
2021/03/20 14:43:53 DEBUG : delluc.mp4: Need to transfer - File not found at Destination
2021/03/20 14:43:53 DEBUG : delluc.mp4: MD5 = c7c32b89b92bcb3b53abcd12b3694146 OK
2021/03/20 14:43:53 INFO  : delluc.mp4: Copied (new) to: 31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4
2021/03/20 14:43:53 INFO  :
Transferred:        6.056M / 6.056 MBytes, 100%, 91.755 MBytes/s, ETA 0s
Transferred:            1 / 1, 100%
Elapsed time:         0.4s

Then the file appears in explorer and plays perfectly, and I can copy it further to GoogleDrive

rclone copy -vv "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4" "TestDrive:baiduyun\Book\160.秋叶Excel数据处理学习班（完结）\第07课：图表篇\【7-2】切片器是动态统计和动态图表/"
2021/03/20 14:49:51 DEBUG : rclone: Version "v1.54-iva01" starting with parameters ["C:\\abyss\\bin\\rclone.exe" "copy" "-vv" "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4" "TestDrive:baiduyun\\Book\\160.秋叶Excel数据处理学习班（完结）\\第07课：图表篇\\【7-2】切片器是动态统计和动态图表/"]
2021/03/20 14:49:51 DEBUG : Creating backend with remote "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4"
2021/03/20 14:49:51 DEBUG : Using config file from "C:\\Users\\ivandeex\\.config\\rclone\\rclone.conf"
2021/03/20 14:49:51 DEBUG : fs cache: adding new entry for parent of "C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4", "//?/C:/BaiduNetdiskDownload/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇"
2021/03/20 14:49:51 DEBUG : Creating backend with remote "TestDrive:baiduyun\\Book\\160.秋叶Excel数据处理学习班（完结）\\第07课：图表篇\\【7-2】切片器是动态统计和动态图表/"
2021/03/20 14:49:52 DEBUG : fs cache: renaming cache item "TestDrive:baiduyun\\Book\\160.秋叶Excel数据处理学习班（完结）\\第07课：图表篇\\【7-2】切片器是动态统计和动态图表/" to be canonical "TestDrive:baiduyun/Book/160.秋叶Excel数据处理学习班（完结）/第07课：图表篇/【7-2】切片器是动态统计和动态图表"
2021/03/20 14:49:52 DEBUG : 31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4: Need to transfer - File not found at Destination
2021/03/20 14:50:02 DEBUG : 31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4: MD5 = c7c32b89b92bcb3b53abcd12b3694146 OK
2021/03/20 14:50:02 INFO  : 31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4: Copied (new)
2021/03/20 14:50:02 INFO  :
Transferred:        6.056M / 6.056 MBytes, 100%, 637.578 kBytes/s, ETA 0s
Transferred:            1 / 1, 100%
Elapsed time:        11.0s

Can't reproduce :-)

**Comment by ivandeex** [comment]
@hagemouse 
Please check that you have permissions to read the source file 请检查您是否具有读取源文件的权限

**Comment by ivandeex** [comment]
@hagemouse 
Can you confirm whether the following commands fail or succeed?

local copy with rclone:
`C:\rclone\rclone.exe copy "C:\BaiduNetdiskDownload\Book\160.秋叶Excel数据处理学习班（完结）\第07课：图表篇\【7-2】切片器是动态统计和动态图表\31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4"  .\`

and native local copy:
`copy "C:\BaiduNetdiskDownload\Book\160.秋叶Excel数据处理学习班（完结）\第07课：图表篇\【7-2】切片器是动态统计和动态图表\31-切片器动态统计和动态图表（）【瑞客论坛 www.ruike1.com】.mp4"  .\`

**Comment by ivandeex** [comment]
Closing stale requests

**Comment by hagemouse** [comment]
Sorry for long time no reply. I just re-installed my computer.
I tested it with rclone 1.55 ,  it still did not work.

copying:
C:\rclone>       copy "C:\BaiduNetdiskDownload\study\女W的Q局[ 公众号：莉莉分享君]\The.Queens.Gambit.S01E06.720p.mp4"   c:\rclone
Copied         1 file.

rclone copy:
C:\rclone>rclone copy "C:\BaiduNetdiskDownload\study\女W的Q局[ 公众号：莉莉分享君]\The.Queens.Gambit.S01E06.720p.mp4"  baidu:                      
2021/04/15 12:51:00 ERROR : : error reading source directory: directory not found
2021/04/15 12:51:00 ERROR : Attempt 1/3 failed with 1 errors and: directory notfound
2021/04/15 12:51:00 ERROR : : error reading source directory: directory not found
2021/04/15 12:51:00 ERROR : Attempt 2/3 failed with 1 errors and: directory not found
2021/04/15 12:51:00 ERROR : : error reading source directory: directory not found
2021/04/15 12:51:00 ERROR : Attempt 3/3 failed with 1 errors and: directory not found
2021/04/15 12:51:00 Failed to copy: directory not found

**Comment by ivandeex** [comment]
I see now.
The problem is with Chinese colon `号：莉`.
rclone actually sees source file but internally converts it to normal colon `:` in a cached copy by default.
Windows fails to create a cache file with colons in the name, hence the confusing `directory not found` error.

As a workaround you can try to add option `--local-encoding=None` recommended by https://github.com/rclone/rclone/issues/5191#issuecomment-812862382, the operation will succeed but be aware that Chinese colon will be replaced by another graphic glyph.

This will be fully fixed in a future rclone version by a fix for #5131

**Comment by ivandeex** [comment]
Related to #5131
Duplicate of #5191
